### PR TITLE
K8SPXC-351: custom ProxySQL config

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPXC-351](https://jira.percona.com/browse/K8SPXC-351)

add functionality to provide own ProxySQL config

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| recreate | passed |
| storage | passed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | failed |
| monitoring | passed |
| self-healing-advanced | failed |
| upgrade-consistency | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | failed |
| storage | failed |
| big-data | failed |
| init-deploy | failed |
| recreate | failed |
| self-healing | failed |
| scaling-proxysql | failed |
| limits | failed |
| restore-to-encrypted-cluster | failed |
| self-healing-advanced | failed |
| upgrade | failed |
| upgrade-consistency | failed |
| operator-self-healing | failed |
| demand-backup | failed |
| monitoring | failed |
| monitoring-2-0 | failed |
| security-context | failed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |
| affinity | failed |
| one-pod | passed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |

File: go.sum [hunk 1]
```diff
@@ -63,8 +63,10 @@ github.com/OneOfOne/xxhash v1.2.2/go.mod h1:HSdplMjZKSmBqAxg5vPj2TmRDmfkzw+cTzAE
 github.com/OneOfOne/xxhash v1.2.6/go.mod h1:eZbhyaAYD41SGSSsnmcpxVoRiQ/MPUTjUdIIOT9Um7Q=
 github.com/PuerkitoBio/purell v1.0.0/go.mod h1:c11w/QuzBsJSee3cPx9rAFu61PvFxuPbtSwDGJws/X0=
 github.com/PuerkitoBio/purell v1.1.0/go.mod h1:c11w/QuzBsJSee3cPx9rAFu61PvFxuPbtSwDGJws/X0=
+github.com/PuerkitoBio/purell v1.1.1 h1:WEQqlqaGbrPkxLJWfBwQmfEAE1Z7ONdDLqrN38tNFfI=
```
Comment (delgod) [inline go.sum:66]: why we need all these libraries for ProxySQL Config?

**Comment by qJkee** [comment]
> I want to ask again :) Did you test it?
> As I can see the conf file is not updated ...
@hors 
Conf file is just an example which could be used in ProxySQL)

File: go.sum (hunk 1 above)
Comment (qJkee) [reply go.sum]: done edd4c2006ee522d324258a2d2086edaedfb71098

**Comment by hors** [comment]
> > I want to ask again :) Did you test it?
> > As I can see the conf file is not updated ...
> 
> @hors
> Conf file is just an example which could be used in ProxySQL)

It does

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| storage | failed |
| big-data | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| self-healing-advanced | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| big-data | passed |
| scaling | passed |
| storage | passed |
| recreate | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| big-data | passed |
| security-context | passed |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| storage | passed |
| affinity | passed |
| self-healing | passed |
| one-pod | passed |
| self-healing-advanced | passed |
| auto-tuning | passed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| big-data | passed |
| limits | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | failed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| big-data | passed |
| storage | passed |
| scaling | passed |
| limits | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring | passed |
| self-healing-advanced | failed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| security-context | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| operator-self-healing | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| big-data | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | failed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup-encrypted-with-tls | passed |

**Pull request merged**